aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -161,6 +163,102 @@ std::tuple<Tensor, Tensor> eig_kernel_impl(const Tensor& self, bool& eigenvector
   return std::tuple<Tensor, Tensor>(vals_, vecs_);
 }
 
+/*
+  Computes eigenvalues and eigenvectors of the input that is stored initially in 'vectors'.
+  The computation is done in-place: 'vectors' stores the input and will be overwritten,
+  'values' should be an allocated empty array.
+  'infos' is used to store information for possible checks for error.
+  'upper' controls the portion of input matrix to consider in computations
+  'compute_eigenvectors' controls whether eigenvectors should be computed.
+  This function doesn't do any error checks and it's assumed that every argument is valid.
+*/
+template <typename scalar_t>
+void apply_lapack_eigh(Tensor& values, Tensor& vectors, Tensor& infos, bool upper, bool compute_eigenvectors) {
+#ifndef USE_LAPACK
+  TORCH_CHECK(
+      false,
+      "Calling torch.linalg.eigh/eigvalsh on a CPU tensor requires compiling ",
+      "PyTorch with LAPACK. Please use PyTorch built with LAPACK support.");
+#else
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+
+  // NumPy allows lowercase input for UPLO argument
```

Comment:
The comment for CUDA suggests we support 'u' and 'l', too. Do we need to add support here?

Reply:
This comment should be removed. I replaced the string UPLO (UPper/LOwer) here with boolean `upper`.